visualization/app/codeCharta/util/algorithm/streetLayout/streetLayoutGenerator.ts

```diff
@@ -0,0 +1,91 @@
+import { CodeMapNode, Node, State, NodeMetricData, BlacklistType, LayoutAlgorithm } from "../../../codeCharta.model"
+import BoundingBox from "./boundingBox"
+import VerticalStreet from "./verticalStreet"
+import HorizontalStreet from "./horizontalStreet"
+import House from "./house"
+import TreeMap from "./treeMap"
+import { Vector2 } from "three"
+import { StreetOrientation } from "./street"
+import { getMapResolutionScaleFactor, isPathBlacklisted, isLeaf } from "../../codeMapHelper"
+import { StreetViewHelper } from "./streetViewHelper"
+import SquarifiedTreeMap from "./squarifiedTreeMap"
+
+const MARGIN_SCALING_FACTOR = 0.02
+const HEIGHT_SCALING_FACTOR = 0.1
+export class StreetLayoutGenerator {
+	static createStreetLayoutNodes(map: CodeMapNode, state: State, metricData: NodeMetricData[], isDeltaState: boolean): Node[] {
+		const mapSizeResolutionScaling = getMapResolutionScaleFactor(state.files)
+		const maxHeight = metricData.find(x => x.name === state.dynamicSettings.heightMetric).maxValue * mapSizeResolutionScaling / HEIGHT_SCALING_FACTOR
+		const heightScale = (state.treeMap.mapSize * 2) / maxHeight
+		
+		const metricName = state.dynamicSettings.areaMetric
+		const mergedMap = StreetViewHelper.mergeDirectories(map, metricName)
+		const maxTreeMapFiles = state.appSettings.maxTreeMapFiles
+		const childBoxes = this.createBoxes(mergedMap, metricName, state, StreetOrientation.Vertical, 0, maxTreeMapFiles)
+		const rootStreet = new HorizontalStreet(mergedMap, childBoxes, 0)
+		rootStreet.calculateDimension(metricName)
+		const margin = state.dynamicSettings.margin * MARGIN_SCALING_FACTOR
+		const layoutNodes: CodeMapNode[] = rootStreet.layout(new Vector2(0, 0), margin)
+		
+		return layoutNodes.map(streetLayoutNode => {
+			return StreetViewHelper.buildNodeFrom(streetLayoutNode as CodeMapNode, heightScale, maxHeight, state, isDeltaState)
```

Comment:
We might want to extend the streetLayoutNode from `CodeMapNode`. That way it should be fine to pass the node through without forcing it.